src/Catel.Core/Catel.Core.Shared/WeakReferences/WeakEventHandler.cs

```diff
@@ -0,0 +1,173 @@
+﻿namespace Catel
```

Comment:
Catel already has WeakEventListener stuff implemented in the core. What's the added value of this versus the WeakEventListener?